// google-ads-node/build/src/index.js
"use strict";
// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.CurrencyConstantServiceClient = exports.ConversionUploadServiceClient = exports.ConversionCustomVariableServiceClient = exports.ConversionAdjustmentUploadServiceClient = exports.ConversionActionServiceClient = exports.CombinedAudienceServiceClient = exports.ClickViewServiceClient = exports.ChangeStatusServiceClient = exports.CarrierConstantServiceClient = exports.CampaignSimulationServiceClient = exports.CampaignSharedSetServiceClient = exports.CampaignServiceClient = exports.CampaignLabelServiceClient = exports.CampaignFeedServiceClient = exports.CampaignExtensionSettingServiceClient = exports.CampaignExperimentServiceClient = exports.CampaignDraftServiceClient = exports.CampaignCriterionSimulationServiceClient = exports.CampaignCriterionServiceClient = exports.CampaignBudgetServiceClient = exports.CampaignBidModifierServiceClient = exports.CampaignAudienceViewServiceClient = exports.CampaignAssetServiceClient = exports.BillingSetupServiceClient = exports.BiddingStrategySimulationServiceClient = exports.BiddingStrategyServiceClient = exports.BatchJobServiceClient = exports.AssetServiceClient = exports.AgeRangeViewServiceClient = exports.AdServiceClient = exports.AdScheduleViewServiceClient = exports.AdParameterServiceClient = exports.AdGroupSimulationServiceClient = exports.AdGroupServiceClient = exports.AdGroupLabelServiceClient = exports.AdGroupFeedServiceClient = exports.AdGroupExtensionSettingServiceClient = exports.AdGroupCriterionSimulationServiceClient = exports.AdGroupCriterionServiceClient = exports.AdGroupCriterionLabelServiceClient = exports.AdGroupBidModifierServiceClient = exports.AdGroupAudienceViewServiceClient = exports.AdGroupAssetServiceClient = exports.AdGroupAdServiceClient = exports.AdGroupAdLabelServiceClient = exports.AdGroupAdAssetViewServiceClient = exports.AccountLinkServiceClient = exports.AccountBudgetServiceClient = exports.AccountBudgetProposalServiceClient = exports.v7 = void 0;
exports.ManagedPlacementViewServiceClient = exports.LocationViewServiceClient = exports.LifeEventServiceClient = exports.LanguageConstantServiceClient = exports.LandingPageViewServiceClient = exports.LabelServiceClient = exports.KeywordViewServiceClient = exports.KeywordPlanServiceClient = exports.KeywordPlanIdeaServiceClient = exports.KeywordPlanCampaignServiceClient = exports.KeywordPlanCampaignKeywordServiceClient = exports.KeywordPlanAdGroupServiceClient = exports.KeywordPlanAdGroupKeywordServiceClient = exports.InvoiceServiceClient = exports.IncomeRangeViewServiceClient = exports.HotelPerformanceViewServiceClient = exports.HotelGroupViewServiceClient = exports.GroupPlacementViewServiceClient = exports.GoogleAdsServiceClient = exports.GoogleAdsFieldServiceClient = exports.GeoTargetConstantServiceClient = exports.GeographicViewServiceClient = exports.GenderViewServiceClient = exports.FeedServiceClient = exports.FeedPlaceholderViewServiceClient = exports.FeedMappingServiceClient = exports.FeedItemTargetServiceClient = exports.FeedItemSetServiceClient = exports.FeedItemSetLinkServiceClient = exports.FeedItemServiceClient = exports.ExtensionFeedItemServiceClient = exports.ExpandedLandingPageViewServiceClient = exports.DynamicSearchAdsSearchTermViewServiceClient = exports.DomainCategoryServiceClient = exports.DistanceViewServiceClient = exports.DisplayKeywordViewServiceClient = exports.DetailPlacementViewServiceClient = exports.CustomInterestServiceClient = exports.CustomerUserAccessServiceClient = exports.CustomerUserAccessInvitationServiceClient = exports.CustomerServiceClient = exports.CustomerNegativeCriterionServiceClient = exports.CustomerManagerLinkServiceClient = exports.CustomerLabelServiceClient = exports.CustomerFeedServiceClient = exports.CustomerExtensionSettingServiceClient = exports.CustomerClientServiceClient = exports.CustomerClientLinkServiceClient = exports.CustomerAssetServiceClient = exports.CustomAudienceServiceClient = void 0;
exports.protos = exports.WebpageViewServiceClient = exports.VideoServiceClient = exports.UserLocationViewServiceClient = exports.UserListServiceClient = exports.UserInterestServiceClient = exports.UserDataServiceClient = exports.TopicViewServiceClient = exports.TopicConstantServiceClient = exports.ThirdPartyAppAnalyticsLinkServiceClient = exports.ShoppingPerformanceViewServiceClient = exports.SharedSetServiceClient = exports.SharedCriterionServiceClient = exports.SearchTermViewServiceClient = exports.RemarketingActionServiceClient = exports.RecommendationServiceClient = exports.ReachPlanServiceClient = exports.ProductGroupViewServiceClient = exports.ProductBiddingCategoryConstantServiceClient = exports.PaymentsAccountServiceClient = exports.ParentalStatusViewServiceClient = exports.PaidOrganicSearchTermViewServiceClient = exports.OperatingSystemVersionConstantServiceClient = exports.OfflineUserDataJobServiceClient = exports.MobileDeviceConstantServiceClient = exports.MobileAppCategoryConstantServiceClient = exports.MerchantCenterLinkServiceClient = exports.MediaFileServiceClient = void 0;
const v7 = require("./v7");
exports.v7 = v7;
const AccountBudgetProposalServiceClient = v7.AccountBudgetProposalServiceClient;
exports.AccountBudgetProposalServiceClient = AccountBudgetProposalServiceClient;
const AccountBudgetServiceClient = v7.AccountBudgetServiceClient;
exports.AccountBudgetServiceClient = AccountBudgetServiceClient;
const AccountLinkServiceClient = v7.AccountLinkServiceClient;
exports.AccountLinkServiceClient = AccountLinkServiceClient;
const AdGroupAdAssetViewServiceClient = v7.AdGroupAdAssetViewServiceClient;
exports.AdGroupAdAssetViewServiceClient = AdGroupAdAssetViewServiceClient;
const AdGroupAdLabelServiceClient = v7.AdGroupAdLabelServiceClient;
exports.AdGroupAdLabelServiceClient = AdGroupAdLabelServiceClient;
const AdGroupAdServiceClient = v7.AdGroupAdServiceClient;
exports.AdGroupAdServiceClient = AdGroupAdServiceClient;
const AdGroupAssetServiceClient = v7.AdGroupAssetServiceClient;
exports.AdGroupAssetServiceClient = AdGroupAssetServiceClient;
const AdGroupAudienceViewServiceClient = v7.AdGroupAudienceViewServiceClient;
exports.AdGroupAudienceViewServiceClient = AdGroupAudienceViewServiceClient;
const AdGroupBidModifierServiceClient = v7.AdGroupBidModifierServiceClient;
exports.AdGroupBidModifierServiceClient = AdGroupBidModifierServiceClient;
const AdGroupCriterionLabelServiceClient = v7.AdGroupCriterionLabelServiceClient;
exports.AdGroupCriterionLabelServiceClient = AdGroupCriterionLabelServiceClient;
const AdGroupCriterionServiceClient = v7.AdGroupCriterionServiceClient;
exports.AdGroupCriterionServiceClient = AdGroupCriterionServiceClient;
const AdGroupCriterionSimulationServiceClient = v7.AdGroupCriterionSimulationServiceClient;
exports.AdGroupCriterionSimulationServiceClient = AdGroupCriterionSimulationServiceClient;
const AdGroupExtensionSettingServiceClient = v7.AdGroupExtensionSettingServiceClient;
exports.AdGroupExtensionSettingServiceClient = AdGroupExtensionSettingServiceClient;
const AdGroupFeedServiceClient = v7.AdGroupFeedServiceClient;
exports.AdGroupFeedServiceClient = AdGroupFeedServiceClient;
const AdGroupLabelServiceClient = v7.AdGroupLabelServiceClient;
exports.AdGroupLabelServiceClient = AdGroupLabelServiceClient;
const AdGroupServiceClient = v7.AdGroupServiceClient;
exports.AdGroupServiceClient = AdGroupServiceClient;
const AdGroupSimulationServiceClient = v7.AdGroupSimulationServiceClient;
exports.AdGroupSimulationServiceClient = AdGroupSimulationServiceClient;
const AdParameterServiceClient = v7.AdParameterServiceClient;
exports.AdParameterServiceClient = AdParameterServiceClient;
const AdScheduleViewServiceClient = v7.AdScheduleViewServiceClient;
exports.AdScheduleViewServiceClient = AdScheduleViewServiceClient;
const AdServiceClient = v7.AdServiceClient;
exports.AdServiceClient = AdServiceClient;
const AgeRangeViewServiceClient = v7.AgeRangeViewServiceClient;
exports.AgeRangeViewServiceClient = AgeRangeViewServiceClient;
const AssetServiceClient = v7.AssetServiceClient;
exports.AssetServiceClient = AssetServiceClient;
const BatchJobServiceClient = v7.BatchJobServiceClient;
exports.BatchJobServiceClient = BatchJobServiceClient;
const BiddingStrategyServiceClient = v7.BiddingStrategyServiceClient;
exports.BiddingStrategyServiceClient = BiddingStrategyServiceClient;
const BiddingStrategySimulationServiceClient = v7.BiddingStrategySimulationServiceClient;
exports.BiddingStrategySimulationServiceClient = BiddingStrategySimulationServiceClient;
const BillingSetupServiceClient = v7.BillingSetupServiceClient;
exports.BillingSetupServiceClient = BillingSetupServiceClient;
const CampaignAssetServiceClient = v7.CampaignAssetServiceClient;
exports.CampaignAssetServiceClient = CampaignAssetServiceClient;
const CampaignAudienceViewServiceClient = v7.CampaignAudienceViewServiceClient;
exports.CampaignAudienceViewServiceClient = CampaignAudienceViewServiceClient;
const CampaignBidModifierServiceClient = v7.CampaignBidModifierServiceClient;
exports.CampaignBidModifierServiceClient = CampaignBidModifierServiceClient;
const CampaignBudgetServiceClient = v7.CampaignBudgetServiceClient;
exports.CampaignBudgetServiceClient = CampaignBudgetServiceClient;
const CampaignCriterionServiceClient = v7.CampaignCriterionServiceClient;
exports.CampaignCriterionServiceClient = CampaignCriterionServiceClient;
const CampaignCriterionSimulationServiceClient = v7.CampaignCriterionSimulationServiceClient;
exports.CampaignCriterionSimulationServiceClient = CampaignCriterionSimulationServiceClient;
const CampaignDraftServiceClient = v7.CampaignDraftServiceClient;
exports.CampaignDraftServiceClient = CampaignDraftServiceClient;
const CampaignExperimentServiceClient = v7.CampaignExperimentServiceClient;
exports.CampaignExperimentServiceClient = CampaignExperimentServiceClient;
const CampaignExtensionSettingServiceClient = v7.CampaignExtensionSettingServiceClient;
exports.CampaignExtensionSettingServiceClient = CampaignExtensionSettingServiceClient;
const CampaignFeedServiceClient = v7.CampaignFeedServiceClient;
exports.CampaignFeedServiceClient = CampaignFeedServiceClient;
const CampaignLabelServiceClient = v7.CampaignLabelServiceClient;
exports.CampaignLabelServiceClient = CampaignLabelServiceClient;
const CampaignServiceClient = v7.CampaignServiceClient;
exports.CampaignServiceClient = CampaignServiceClient;
const CampaignSharedSetServiceClient = v7.CampaignSharedSetServiceClient;
exports.CampaignSharedSetServiceClient = CampaignSharedSetServiceClient;
const CampaignSimulationServiceClient = v7.CampaignSimulationServiceClient;
exports.CampaignSimulationServiceClient = CampaignSimulationServiceClient;
const CarrierConstantServiceClient = v7.CarrierConstantServiceClient;
exports.CarrierConstantServiceClient = CarrierConstantServiceClient;
const ChangeStatusServiceClient = v7.ChangeStatusServiceClient;
exports.ChangeStatusServiceClient = ChangeStatusServiceClient;
const ClickViewServiceClient = v7.ClickViewServiceClient;
exports.ClickViewServiceClient = ClickViewServiceClient;
const CombinedAudienceServiceClient = v7.CombinedAudienceServiceClient;
exports.CombinedAudienceServiceClient = CombinedAudienceServiceClient;
const ConversionActionServiceClient = v7.ConversionActionServiceClient;
exports.ConversionActionServiceClient = ConversionActionServiceClient;
const ConversionAdjustmentUploadServiceClient = v7.ConversionAdjustmentUploadServiceClient;
exports.ConversionAdjustmentUploadServiceClient = ConversionAdjustmentUploadServiceClient;
const ConversionCustomVariableServiceClient = v7.ConversionCustomVariableServiceClient;
exports.ConversionCustomVariableServiceClient = ConversionCustomVariableServiceClient;
const ConversionUploadServiceClient = v7.ConversionUploadServiceClient;
exports.ConversionUploadServiceClient = ConversionUploadServiceClient;
const CurrencyConstantServiceClient = v7.CurrencyConstantServiceClient;
exports.CurrencyConstantServiceClient = CurrencyConstantServiceClient;
const CustomAudienceServiceClient = v7.CustomAudienceServiceClient;
exports.CustomAudienceServiceClient = CustomAudienceServiceClient;
const CustomerAssetServiceClient = v7.CustomerAssetServiceClient;
exports.CustomerAssetServiceClient = CustomerAssetServiceClient;
const CustomerClientLinkServiceClient = v7.CustomerClientLinkServiceClient;
exports.CustomerClientLinkServiceClient = CustomerClientLinkServiceClient;
const CustomerClientServiceClient = v7.CustomerClientServiceClient;
exports.CustomerClientServiceClient = CustomerClientServiceClient;
const CustomerExtensionSettingServiceClient = v7.CustomerExtensionSettingServiceClient;
exports.CustomerExtensionSettingServiceClient = CustomerExtensionSettingServiceClient;
const CustomerFeedServiceClient = v7.CustomerFeedServiceClient;
exports.CustomerFeedServiceClient = CustomerFeedServiceClient;
const CustomerLabelServiceClient = v7.CustomerLabelServiceClient;
exports.CustomerLabelServiceClient = CustomerLabelServiceClient;
const CustomerManagerLinkServiceClient = v7.CustomerManagerLinkServiceClient;
exports.CustomerManagerLinkServiceClient = CustomerManagerLinkServiceClient;
const CustomerNegativeCriterionServiceClient = v7.CustomerNegativeCriterionServiceClient;
exports.CustomerNegativeCriterionServiceClient = CustomerNegativeCriterionServiceClient;
const CustomerServiceClient = v7.CustomerServiceClient;
exports.CustomerServiceClient = CustomerServiceClient;
const CustomerUserAccessInvitationServiceClient = v7.CustomerUserAccessInvitationServiceClient;
exports.CustomerUserAccessInvitationServiceClient = CustomerUserAccessInvitationServiceClient;
const CustomerUserAccessServiceClient = v7.CustomerUserAccessServiceClient;
exports.CustomerUserAccessServiceClient = CustomerUserAccessServiceClient;
const CustomInterestServiceClient = v7.CustomInterestServiceClient;
exports.CustomInterestServiceClient = CustomInterestServiceClient;
const DetailPlacementViewServiceClient = v7.DetailPlacementViewServiceClient;
exports.DetailPlacementViewServiceClient = DetailPlacementViewServiceClient;
const DisplayKeywordViewServiceClient = v7.DisplayKeywordViewServiceClient;
exports.DisplayKeywordViewServiceClient = DisplayKeywordViewServiceClient;
const DistanceViewServiceClient = v7.DistanceViewServiceClient;
exports.DistanceViewServiceClient = DistanceViewServiceClient;
const DomainCategoryServiceClient = v7.DomainCategoryServiceClient;
exports.DomainCategoryServiceClient = DomainCategoryServiceClient;
const DynamicSearchAdsSearchTermViewServiceClient = v7.DynamicSearchAdsSearchTermViewServiceClient;
exports.DynamicSearchAdsSearchTermViewServiceClient = DynamicSearchAdsSearchTermViewServiceClient;
const ExpandedLandingPageViewServiceClient = v7.ExpandedLandingPageViewServiceClient;
exports.ExpandedLandingPageViewServiceClient = ExpandedLandingPageViewServiceClient;
const ExtensionFeedItemServiceClient = v7.ExtensionFeedItemServiceClient;
exports.ExtensionFeedItemServiceClient = ExtensionFeedItemServiceClient;
const FeedItemServiceClient = v7.FeedItemServiceClient;
exports.FeedItemServiceClient = FeedItemServiceClient;
const FeedItemSetLinkServiceClient = v7.FeedItemSetLinkServiceClient;
exports.FeedItemSetLinkServiceClient = FeedItemSetLinkServiceClient;
const FeedItemSetServiceClient = v7.FeedItemSetServiceClient;
exports.FeedItemSetServiceClient = FeedItemSetServiceClient;
const FeedItemTargetServiceClient = v7.FeedItemTargetServiceClient;
exports.FeedItemTargetServiceClient = FeedItemTargetServiceClient;
const FeedMappingServiceClient = v7.FeedMappingServiceClient;
exports.FeedMappingServiceClient = FeedMappingServiceClient;
const FeedPlaceholderViewServiceClient = v7.FeedPlaceholderViewServiceClient;
exports.FeedPlaceholderViewServiceClient = FeedPlaceholderViewServiceClient;
const FeedServiceClient = v7.FeedServiceClient;
exports.FeedServiceClient = FeedServiceClient;
const GenderViewServiceClient = v7.GenderViewServiceClient;
exports.GenderViewServiceClient = GenderViewServiceClient;
const GeographicViewServiceClient = v7.GeographicViewServiceClient;
exports.GeographicViewServiceClient = GeographicViewServiceClient;
const GeoTargetConstantServiceClient = v7.GeoTargetConstantServiceClient;
exports.GeoTargetConstantServiceClient = GeoTargetConstantServiceClient;
const GoogleAdsFieldServiceClient = v7.GoogleAdsFieldServiceClient;
exports.GoogleAdsFieldServiceClient = GoogleAdsFieldServiceClient;
const GoogleAdsServiceClient = v7.GoogleAdsServiceClient;
exports.GoogleAdsServiceClient = GoogleAdsServiceClient;
const GroupPlacementViewServiceClient = v7.GroupPlacementViewServiceClient;
exports.GroupPlacementViewServiceClient = GroupPlacementViewServiceClient;
const HotelGroupViewServiceClient = v7.HotelGroupViewServiceClient;
exports.HotelGroupViewServiceClient = HotelGroupViewServiceClient;
const HotelPerformanceViewServiceClient = v7.HotelPerformanceViewServiceClient;
exports.HotelPerformanceViewServiceClient = HotelPerformanceViewServiceClient;
const IncomeRangeViewServiceClient = v7.IncomeRangeViewServiceClient;
exports.IncomeRangeViewServiceClient = IncomeRangeViewServiceClient;
const InvoiceServiceClient = v7.InvoiceServiceClient;
exports.InvoiceServiceClient = InvoiceServiceClient;
const KeywordPlanAdGroupKeywordServiceClient = v7.KeywordPlanAdGroupKeywordServiceClient;
exports.KeywordPlanAdGroupKeywordServiceClient = KeywordPlanAdGroupKeywordServiceClient;
const KeywordPlanAdGroupServiceClient = v7.KeywordPlanAdGroupServiceClient;
exports.KeywordPlanAdGroupServiceClient = KeywordPlanAdGroupServiceClient;
const KeywordPlanCampaignKeywordServiceClient = v7.KeywordPlanCampaignKeywordServiceClient;
exports.KeywordPlanCampaignKeywordServiceClient = KeywordPlanCampaignKeywordServiceClient;
const KeywordPlanCampaignServiceClient = v7.KeywordPlanCampaignServiceClient;
exports.KeywordPlanCampaignServiceClient = KeywordPlanCampaignServiceClient;
const KeywordPlanIdeaServiceClient = v7.KeywordPlanIdeaServiceClient;
exports.KeywordPlanIdeaServiceClient = KeywordPlanIdeaServiceClient;
const KeywordPlanServiceClient = v7.KeywordPlanServiceClient;
exports.KeywordPlanServiceClient = KeywordPlanServiceClient;
const KeywordViewServiceClient = v7.KeywordViewServiceClient;
exports.KeywordViewServiceClient = KeywordViewServiceClient;
const LabelServiceClient = v7.LabelServiceClient;
exports.LabelServiceClient = LabelServiceClient;
const LandingPageViewServiceClient = v7.LandingPageViewServiceClient;
exports.LandingPageViewServiceClient = LandingPageViewServiceClient;
const LanguageConstantServiceClient = v7.LanguageConstantServiceClient;
exports.LanguageConstantServiceClient = LanguageConstantServiceClient;
const LifeEventServiceClient = v7.LifeEventServiceClient;
exports.LifeEventServiceClient = LifeEventServiceClient;
const LocationViewServiceClient = v7.LocationViewServiceClient;
exports.LocationViewServiceClient = LocationViewServiceClient;
const ManagedPlacementViewServiceClient = v7.ManagedPlacementViewServiceClient;
exports.ManagedPlacementViewServiceClient = ManagedPlacementViewServiceClient;
const MediaFileServiceClient = v7.MediaFileServiceClient;
exports.MediaFileServiceClient = MediaFileServiceClient;
const MerchantCenterLinkServiceClient = v7.MerchantCenterLinkServiceClient;
exports.MerchantCenterLinkServiceClient = MerchantCenterLinkServiceClient;
const MobileAppCategoryConstantServiceClient = v7.MobileAppCategoryConstantServiceClient;
exports.MobileAppCategoryConstantServiceClient = MobileAppCategoryConstantServiceClient;
const MobileDeviceConstantServiceClient = v7.MobileDeviceConstantServiceClient;
exports.MobileDeviceConstantServiceClient = MobileDeviceConstantServiceClient;
const OfflineUserDataJobServiceClient = v7.OfflineUserDataJobServiceClient;
exports.OfflineUserDataJobServiceClient = OfflineUserDataJobServiceClient;
const OperatingSystemVersionConstantServiceClient = v7.OperatingSystemVersionConstantServiceClient;
exports.OperatingSystemVersionConstantServiceClient = OperatingSystemVersionConstantServiceClient;
const PaidOrganicSearchTermViewServiceClient = v7.PaidOrganicSearchTermViewServiceClient;
exports.PaidOrganicSearchTermViewServiceClient = PaidOrganicSearchTermViewServiceClient;
const ParentalStatusViewServiceClient = v7.ParentalStatusViewServiceClient;
exports.ParentalStatusViewServiceClient = ParentalStatusViewServiceClient;
const PaymentsAccountServiceClient = v7.PaymentsAccountServiceClient;
exports.PaymentsAccountServiceClient = PaymentsAccountServiceClient;
const ProductBiddingCategoryConstantServiceClient = v7.ProductBiddingCategoryConstantServiceClient;
exports.ProductBiddingCategoryConstantServiceClient = ProductBiddingCategoryConstantServiceClient;
const ProductGroupViewServiceClient = v7.ProductGroupViewServiceClient;
exports.ProductGroupViewServiceClient = ProductGroupViewServiceClient;
const ReachPlanServiceClient = v7.ReachPlanServiceClient;
exports.ReachPlanServiceClient = ReachPlanServiceClient;
const RecommendationServiceClient = v7.RecommendationServiceClient;
exports.RecommendationServiceClient = RecommendationServiceClient;
const RemarketingActionServiceClient = v7.RemarketingActionServiceClient;
exports.RemarketingActionServiceClient = RemarketingActionServiceClient;
const SearchTermViewServiceClient = v7.SearchTermViewServiceClient;
exports.SearchTermViewServiceClient = SearchTermViewServiceClient;
const SharedCriterionServiceClient = v7.SharedCriterionServiceClient;
exports.SharedCriterionServiceClient = SharedCriterionServiceClient;
const SharedSetServiceClient = v7.SharedSetServiceClient;
exports.SharedSetServiceClient = SharedSetServiceClient;
const ShoppingPerformanceViewServiceClient = v7.ShoppingPerformanceViewServiceClient;
exports.ShoppingPerformanceViewServiceClient = ShoppingPerformanceViewServiceClient;
const ThirdPartyAppAnalyticsLinkServiceClient = v7.ThirdPartyAppAnalyticsLinkServiceClient;
exports.ThirdPartyAppAnalyticsLinkServiceClient = ThirdPartyAppAnalyticsLinkServiceClient;
const TopicConstantServiceClient = v7.TopicConstantServiceClient;
exports.TopicConstantServiceClient = TopicConstantServiceClient;
const TopicViewServiceClient = v7.TopicViewServiceClient;
exports.TopicViewServiceClient = TopicViewServiceClient;
const UserDataServiceClient = v7.UserDataServiceClient;
exports.UserDataServiceClient = UserDataServiceClient;
const UserInterestServiceClient = v7.UserInterestServiceClient;
exports.UserInterestServiceClient = UserInterestServiceClient;
const UserListServiceClient = v7.UserListServiceClient;
exports.UserListServiceClient = UserListServiceClient;
const UserLocationViewServiceClient = v7.UserLocationViewServiceClient;
exports.UserLocationViewServiceClient = UserLocationViewServiceClient;
const VideoServiceClient = v7.VideoServiceClient;
exports.VideoServiceClient = VideoServiceClient;
const WebpageViewServiceClient = v7.WebpageViewServiceClient;
exports.WebpageViewServiceClient = WebpageViewServiceClient;
exports.default = { v7, AccountBudgetProposalServiceClient, AccountBudgetServiceClient, AccountLinkServiceClient, AdGroupAdAssetViewServiceClient, AdGroupAdLabelServiceClient, AdGroupAdServiceClient, AdGroupAssetServiceClient, AdGroupAudienceViewServiceClient, AdGroupBidModifierServiceClient, AdGroupCriterionLabelServiceClient, AdGroupCriterionServiceClient, AdGroupCriterionSimulationServiceClient, AdGroupExtensionSettingServiceClient, AdGroupFeedServiceClient, AdGroupLabelServiceClient, AdGroupServiceClient, AdGroupSimulationServiceClient, AdParameterServiceClient, AdScheduleViewServiceClient, AdServiceClient, AgeRangeViewServiceClient, AssetServiceClient, BatchJobServiceClient, BiddingStrategyServiceClient, BiddingStrategySimulationServiceClient, BillingSetupServiceClient, CampaignAssetServiceClient, CampaignAudienceViewServiceClient, CampaignBidModifierServiceClient, CampaignBudgetServiceClient, CampaignCriterionServiceClient, CampaignCriterionSimulationServiceClient, CampaignDraftServiceClient, CampaignExperimentServiceClient, CampaignExtensionSettingServiceClient, CampaignFeedServiceClient, CampaignLabelServiceClient, CampaignServiceClient, CampaignSharedSetServiceClient, CampaignSimulationServiceClient, CarrierConstantServiceClient, ChangeStatusServiceClient, ClickViewServiceClient, CombinedAudienceServiceClient, ConversionActionServiceClient, ConversionAdjustmentUploadServiceClient, ConversionCustomVariableServiceClient, ConversionUploadServiceClient, CurrencyConstantServiceClient, CustomAudienceServiceClient, CustomerAssetServiceClient, CustomerClientLinkServiceClient, CustomerClientServiceClient, CustomerExtensionSettingServiceClient, CustomerFeedServiceClient, CustomerLabelServiceClient, CustomerManagerLinkServiceClient, CustomerNegativeCriterionServiceClient, CustomerServiceClient, CustomerUserAccessInvitationServiceClient, CustomerUserAccessServiceClient, CustomInterestServiceClient, DetailPlacementViewServiceClient, DisplayKeywordViewServiceClient, DistanceViewServiceClient, DomainCategoryServiceClient, DynamicSearchAdsSearchTermViewServiceClient, ExpandedLandingPageViewServiceClient, ExtensionFeedItemServiceClient, FeedItemServiceClient, FeedItemSetLinkServiceClient, FeedItemSetServiceClient, FeedItemTargetServiceClient, FeedMappingServiceClient, FeedPlaceholderViewServiceClient, FeedServiceClient, GenderViewServiceClient, GeographicViewServiceClient, GeoTargetConstantServiceClient, GoogleAdsFieldServiceClient, GoogleAdsServiceClient, GroupPlacementViewServiceClient, HotelGroupViewServiceClient, HotelPerformanceViewServiceClient, IncomeRangeViewServiceClient, InvoiceServiceClient, KeywordPlanAdGroupKeywordServiceClient, KeywordPlanAdGroupServiceClient, KeywordPlanCampaignKeywordServiceClient, KeywordPlanCampaignServiceClient, KeywordPlanIdeaServiceClient, KeywordPlanServiceClient, KeywordViewServiceClient, LabelServiceClient, LandingPageViewServiceClient, LanguageConstantServiceClient, LifeEventServiceClient, LocationViewServiceClient, ManagedPlacementViewServiceClient, MediaFileServiceClient, MerchantCenterLinkServiceClient, MobileAppCategoryConstantServiceClient, MobileDeviceConstantServiceClient, OfflineUserDataJobServiceClient, OperatingSystemVersionConstantServiceClient, PaidOrganicSearchTermViewServiceClient, ParentalStatusViewServiceClient, PaymentsAccountServiceClient, ProductBiddingCategoryConstantServiceClient, ProductGroupViewServiceClient, ReachPlanServiceClient, RecommendationServiceClient, RemarketingActionServiceClient, SearchTermViewServiceClient, SharedCriterionServiceClient, SharedSetServiceClient, ShoppingPerformanceViewServiceClient, ThirdPartyAppAnalyticsLinkServiceClient, TopicConstantServiceClient, TopicViewServiceClient, UserDataServiceClient, UserInterestServiceClient, UserListServiceClient, UserLocationViewServiceClient, VideoServiceClient, WebpageViewServiceClient };
const protos = require("../protos/protos");
exports.protos = protos;
